packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageStylesComfyV2.ts

```diff
@@ -0,0 +1,227 @@
+import { isNil } from 'lodash';
+import { ComponentSlotStylesPrepared, ICSSInJSStyle } from '@fluentui/styles';
+import { getBorderFocusStyles } from '../../getBorderFocusStyles';
+import { chatMessageSlotClassNames, ChatMessageStylesProps } from '../../../../components/Chat/ChatMessage';
+import { pxToRem } from '../../../../utils';
+import { screenReaderContainerStyles } from '../../../../utils/accessibility/Styles/accessibilityStyles';
+import { ChatMessageVariables } from './chatMessageVariables';
+
+const displayActionMenu = (overlayZIndex: ICSSInJSStyle['zIndex']): ICSSInJSStyle => ({
+  zIndex: overlayZIndex!,
+  overflow: 'visible',
+  opacity: 1,
+  width: 'auto',
+});
+
+export const chatMessageStylesComfyV2: ComponentSlotStylesPrepared<ChatMessageStylesProps, ChatMessageVariables> = {
+  root: ({ props: p, variables: v, theme }): ICSSInJSStyle => {
+    const borderFocusStyles = getBorderFocusStyles({
+      borderRadius: 'inherit',
+      variables: theme.siteVariables,
+    });
+
+    // Fixes the bubble focus border rendering on top of the user avatar
+    delete (borderFocusStyles[':focus-visible'][':before'] as ICSSInJSStyle).zIndex;
+    delete (borderFocusStyles[':focus-visible'][':after'] as ICSSInJSStyle).zIndex;
+
+    return {
+      display: 'flex',
+      flexDirection: 'column',
+      alignItems: p.mine ? 'flex-end' : 'flex-start',
+      outline: 'none',
+
+      [`&:focus-visible .${chatMessageSlotClassNames.timestamp}`]: {
+        display: 'block',
+      },
+
+      [`&:focus-visible .${chatMessageSlotClassNames.bubble}`]: borderFocusStyles[':focus-visible'],
+
+      [`& .bubble-inset:hover .${chatMessageSlotClassNames.timestamp}`]: {
+        display: 'block',
+      },
+    };
+  },
+
+  header: ({ props: p, theme }): ICSSInJSStyle => ({
+    display: 'flex',
+    width: '100%',
+    justifyContent: p.mine ? 'flex-end' : 'start',
+    gap: pxToRem(8),
+    '& > div': {
+      paddingTop: pxToRem(8),
+    },
+    color: theme.siteVariables.colorScheme.default.foreground2,
+  }),
+
+  author: ({ props: p }): ICSSInJSStyle => ({
+    ...((p.mine || p.attached === 'bottom' || p.attached === true) && (screenReaderContainerStyles as ICSSInJSStyle)),
+    fontWeight: 400,
+    marginBottom: pxToRem(2),
+    overflow: 'hidden',
+    whiteSpace: 'nowrap',
+    textOverflow: 'ellipsis',
+  }),
+
+  timestamp: ({ variables: v }): ICSSInJSStyle => {
+    const styles: ICSSInJSStyle = {
+      display: 'none',
+      whiteSpace: 'nowrap',
+    };
+    if (v.v2_isNarrow) {
+      styles.fontSize = '1rem';
+      styles.marginTop = pxToRem(3);
+      styles.marginLeft = pxToRem(2.5);
+      styles.marginRight = pxToRem(2.5);
+      styles.marginBottom = 0;
+    }
+    return styles;
+  },
+
+  body: ({ props: p, variables: v }): ICSSInJSStyle => ({
+    position: 'relative',
+    maxWidth: '100%',
+    ...(v.v2_isFullWidth && {
```

Comment:
Where is this variable set?

Reply:
Userland (or in examples that I need to write).

Reply:
May be better as a prop, if kept at all.